BUG: Hypothesis test failure - test_on_offset_implementations

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=37199&view=logs&j=76104ccd-8dcc-5006-a17c-28bcdd709542&t=cd420693-444f-5c55-14e0-1be238e189a1

```
2020-06-13T14:27:46.0206662Z ================================== FAILURES ===================================
2020-06-13T14:27:46.0207049Z _______________________ test_on_offset_implementations ________________________
2020-06-13T14:27:46.0207455Z [gw2] win32 -- Python 3.7.7 C:\Miniconda\envs\pandas-dev\python.exe
2020-06-13T14:27:46.0207694Z 
2020-06-13T14:27:46.0207905Z     @given(gen_random_datetime, gen_yqm_offset)
2020-06-13T14:27:46.0208239Z >   def test_on_offset_implementations(dt, offset):
2020-06-13T14:27:46.0208612Z 
2020-06-13T14:27:46.0208902Z pandas\tests\tseries\offsets\test_offsets_properties.py:89: 
2020-06-13T14:27:46.0209414Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-06-13T14:27:46.0209969Z pandas\tests\tseries\offsets\test_offsets_properties.py:94: in test_on_offset_implementations
2020-06-13T14:27:46.0210468Z     compare = (dt + offset) - offset
2020-06-13T14:27:46.0210950Z pandas\_libs\tslibs\offsets.pyx:438: in pandas._libs.tslibs.offsets.BaseOffset.__add__
2020-06-13T14:27:46.0211305Z     return other.__add__(self)
2020-06-13T14:27:46.0211774Z pandas\_libs\tslibs\offsets.pyx:440: in pandas._libs.tslibs.offsets.BaseOffset.__add__
2020-06-13T14:27:46.0212253Z     return self.apply(other)
2020-06-13T14:27:46.0212687Z pandas\_libs\tslibs\offsets.pyx:135: in pandas._libs.tslibs.offsets.apply_wraps.wrapper
2020-06-13T14:27:46.0213172Z     result = result.tz_localize(tz)
2020-06-13T14:27:46.0213673Z pandas\_libs\tslibs\timestamps.pyx:1258: in pandas._libs.tslibs.timestamps.Timestamp.tz_localize
2020-06-13T14:27:46.0215173Z     value = tz_localize_to_utc(np.array([self.value], dtype='i8'), tz,
2020-06-13T14:27:46.0215769Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
2020-06-13T14:27:46.0216156Z 
2020-06-13T14:27:46.0216385Z >   raise pytz.NonExistentTimeError(stamp)
2020-06-13T14:27:46.0216849Z E   pytz.exceptions.NonExistentTimeError: 1906-01-01 00:00:00
2020-06-13T14:27:46.0217233Z 
2020-06-13T14:27:46.0217489Z pandas\_libs\tslibs\tzconversion.pyx:276: NonExistentTimeError
2020-06-13T14:27:46.0217864Z --------------------------------- Hypothesis ----------------------------------
2020-06-13T14:27:46.0218223Z Falsifying example: test_on_offset_implementations(
2020-06-13T14:27:46.0218570Z     dt=datetime.datetime(1900, 1, 1, 0, 0, tzinfo=tzfile('Asia/Calcutta')),
2020-06-13T14:27:46.0219018Z     offset=<72 * MonthBegins>,
2020-06-13T14:27:46.0219213Z )
```

Getting same error.

I think it is because the test adds an `offset` to a randomly generated date. If the randomly generated date is close `max_dt` adding the `offset` pushes it over the edge so it is not a valid value. 

https://github.com/pandas-dev/pandas/blob/master/pandas/tests/tseries/offsets/test_offsets_properties.py#L93


https://github.com/pandas-dev/pandas/blob/master/pandas/tests/tseries/offsets/test_offsets_properties.py#L56


https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=39375&view=logs&j=077026cf-93c0-54aa-45e0-9996ba75f6f7&t=e95cf409-86ae-5b4d-6c5f-79395ef75e8f

```
================================== FAILURES ===================================
_______________________ test_on_offset_implementations ________________________
[gw0] win32 -- Python 3.6.10 C:\Miniconda\envs\pandas-dev\python.exe

    @given(gen_random_datetime, gen_yqm_offset)
>   def test_on_offset_implementations(dt, offset):

pandas\tests\tseries\offsets\test_offsets_properties.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\tests\tseries\offsets\test_offsets_properties.py:93: in test_on_offset_implementations
    compare = (dt + offset) - offset
pandas\_libs\tslibs\offsets.pyx:407: in pandas._libs.tslibs.offsets.BaseOffset.__add__
    return other.__add__(self)
pandas\_libs\tslibs\offsets.pyx:409: in pandas._libs.tslibs.offsets.BaseOffset.__add__
    return self.apply(other)
pandas\_libs\tslibs\offsets.pyx:157: in pandas._libs.tslibs.offsets.apply_wraps.wrapper
    result = result.tz_localize(tz)
pandas\_libs\tslibs\timestamps.pyx:1305: in pandas._libs.tslibs.timestamps.Timestamp.tz_localize
    value = tz_localize_to_utc_single(self.value, tz,
pandas\_libs\tslibs\tzconversion.pyx:57: in pandas._libs.tslibs.tzconversion.tz_localize_to_utc_single
    return tz_localize_to_utc(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise pytz.NonExistentTimeError(stamp)
```

--- a/pandas/tests/tseries/offsets/test_offsets_properties.py
+++ b/pandas/tests/tseries/offsets/test_offsets_properties.py
@@ -13,6 +13,7 @@ from hypothesis import assume, given, strategies as st
 from hypothesis.extra.dateutil import timezones as dateutil_timezones
 from hypothesis.extra.pytz import timezones as pytz_timezones
 import pytest
+import pytz
 
 import pandas as pd
 from pandas import Timestamp
@@ -92,7 +93,13 @@ def test_on_offset_implementations(dt, offset):
     # check that the class-specific implementations of is_on_offset match
     # the general case definition:
     #   (dt + offset) - offset == dt
-    compare = (dt + offset) - offset
+    try:
+        compare = (dt + offset) - offset
+    except pytz.NonExistentTimeError:
+        # dt + offset does not exist, assume(False) to indicate
+        #  to hypothesis that this is not a valid test case
+        assume(False)
+
     assert offset.is_on_offset(dt) == (compare == dt)
 
 

